sklearn/decomposition/_kernel_pca.py

```diff
@@ -335,10 +341,10 @@ def transform(self, X):
         K = self._centerer.transform(self._get_kernel(X, self.X_fit_))
 
         # scale eigenvectors (properly account for null-space for dot product)
-        non_zeros = np.flatnonzero(self.lambdas_)
-        scaled_alphas = np.zeros_like(self.alphas_)
-        scaled_alphas[:, non_zeros] = (self.alphas_[:, non_zeros]
-                                       / np.sqrt(self.lambdas_[non_zeros]))
+        non_zeros = np.flatnonzero(self.eigenvalues_)
+        scaled_alphas = np.zeros_like(self.eigenvectors_)
+        scaled_alphas[:, non_zeros] = (self.eigenvectors_[:, non_zeros]
+                                       / np.sqrt(self.eigenvalues_[non_zeros]))
 
         # Project with a scalar product between K and the scaled eigenvectors
         return np.dot(K, scaled_alphas)
```

Comment:
Even though `scaled_alphas` is not publicly exposed, it can be renamed as well for consistency:
```suggestion
        scaled_eigenvectors = np.zeros_like(self.eigenvectors_)
        scaled_eigenvectors[:, non_zeros] = (self.eigenvectors_[:, non_zeros]
                                             / np.sqrt(self.eigenvalues_[non_zeros]))

        # Project with a scalar product between K and the scaled eigenvectors
        return np.dot(K, scaled_eigenvectors)
```

Reply:
Thanks! That's a good point!